extensions/ql-vscode/src/queries.ts

```diff
@@ -273,7 +280,13 @@ async function checkAndConfirmDatabaseUpgrade(qs: qsClient.QueryServerClient, db
 
   logger.log(descriptionMessage);
   // Ask the user to confirm the upgrade.
-  const shouldUpgrade = await helpers.showBinaryChoiceDialog(`Should the database ${db.databaseUri.fsPath} be upgraded?\n\n${descriptionMessage}`);
+
+  let messageLines = descriptionMessage.split('\n');
+  if (messageLines.length > MAX_MESSAGE_LINES) {
```

Comment:
Nit: per the JSDoc, I think this should be `MAX_MESSAGE_LINES - 2` since we add two extra lines in line 289.